### PR TITLE
Add metrics for creation of products and assignments

### PR DESCRIPTION
**Pull request by egriff38ford**
## Issue
Part of #471 - Determine Metrics Strategy

## What was done
- [x] Added product creation to tracked events
- [x] Added product assignments to tracked events
- [x] Added helper to aid in creating more events in the future

## How to test
1. Create a product
2. Assign somebody to the product
3. See that those actions wind up in [Matomo](https://matomo-fordlabs.apps.pd01i.edc1.cf.ford.com/)

Feature URL: https://featurebranchui.apps.pd01i.edc1.cf.ford.com/
Feature Branch Deployment: https://jenkins-fordlabs.apps.pd01e.edc1.cf.ford.com/job/PeopleMover/job/Deploy%20PeopleMover%20FeatureBranch/

### CODE REVIEW

File: ui/src/Products/ProductClient.test.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import ProductClient from "./ProductClient";
+import {Product} from "./Product";
+import axios, {AxiosPromise} from 'axios';
+import {MatomoWindow} from "../CommonTypes/MatomoWindow";
+import AssignmentClient from "../Assignments/AssignmentClient";
+import TestUtils from "../tests/TestUtils";
```
Comment (RandomAshes) [inline ui/src/Products/ProductClient.test.ts:6]: More double quotes

File: ui/src/Products/ProductClient.test.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        originalWindow = window;
+    });
+
+    afterEach(() => {
+        (window as Window) = originalWindow;
+    });
+
+    it('should push create product action on create', async () => {
+        let expectedName = "Floam";
+        let product: Product = {
```
Comment (RandomAshes) [inline ui/src/Products/ProductClient.test.ts:25]: These all can be const

File: ui/src/Products/ProductClient.test.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            spaceId: 0,
+        };
+
+        let expectedResponse = {code: 417};
+        axios.post = jest.fn(() => Promise.reject(expectedResponse as any));
+
+        try {
+            await ProductClient.createProduct("a uuid", product);
+        } catch(err)  {
+            expect(window._paq).toContainEqual(['trackEvent', 'Product', 'createError', expectedName, expectedResponse.code])
```
Comment (RandomAshes) [inline ui/src/Products/ProductClient.test.ts:62]: To make sure it's getting inside this catch, I'd suggest using a done callback here.

File: ui/src/Products/ProductClient.test.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        let axiosResponse = await ProductClient.createProduct("a uuid", product);
+
+        expect(axiosResponse).toBe(expectedResponse);
+
+        expect(window._paq).toContainEqual(['trackEvent', 'Product', 'create', expectedName])
+
+    });
+
+    it('should push createError on create with failure code', async () => {
+        let expectedName = "Floam";
```
Comment (RandomAshes) [inline ui/src/Products/ProductClient.test.ts:46]: const and single quotes

File: ui/src/Auth/TokenProvider.tsx [hunk 5]
```diff
@@ -28,16 +31,26 @@ export const removeToken = (): void => {
     cookie.remove('accessToken', {path: '/'});
 };
 
-export const getDecodedToken = (): unknown | null  => {
+export const getDecodedToken = (): unknown | null => {
     const accessToken = getToken();
     return jwtDecoder(accessToken);
 };
 
+const addUserToMatomo = (userName: string) => {
```
Comment (RandomAshes) [inline ui/src/Auth/TokenProvider.tsx:39]: Should this method be in Matomo events?

Comment (egriff38ford) [reply ui/src/Auth/TokenProvider.tsx:39]: This only happens once, so probably not necessary?

File: ui/src/Products/ProductClient.test.ts (hunk 3 above)
Comment (egriff38ford) [reply ui/src/Products/ProductClient.test.ts:62]: We're waiting for the response, so it should get into there

File: ui/src/Auth/TokenProvider.tsx (hunk 5 above)
Comment (RandomAshes) [reply ui/src/Auth/TokenProvider.tsx:39]: Mike and I agree it makes more sense to keep all the matomo logic together if we can. I'd hate for someone to need this method again and not realize it already exists and where it exists.

**Review by sabdul14 — APPROVED** [review]

**Pull request merged**